state/accountsDB_test.go

```diff
@@ -432,6 +469,298 @@ func TestAccountsDB_SaveAccountWithSomeValuesShouldWork(t *testing.T) {
+				}
+
+				getCalls++
+				return common.NewTrieLeafHolder(nil, 0, core.NotSpecified), nil
+			},
+			UpdateCalled: func(key, value []byte) error {
+				return nil
+			},
+			UpdateWithVersionCalled: func(key, value []byte, version core.TrieNodeVersion) error {
+				return nil
+			},
+			GetStorageManagerCalled: func() common.StorageManager {
+				return &storageManager.StorageManagerStub{
+					PutCalled: func(key, value []byte) error {
+						putCalled = true
+						require.Equal(t, key, codeDataHash)
+						require.Equal(t, value, codeData)
+						return nil
+					},
+				}
+			},
+		}
+
+		args.Trie = ts
+
+		adb, err := state.NewAccountsDB(args)
+		require.Nil(t, err)
+
+		adr = make([]byte, 32)
+		account := stateMock.NewAccountWrapMock(adr)
+		account.SetCode(codeData)
+		account.SetVersion(uint8(core.WithoutCodeLeaf))
+
+		err = adb.SaveAccount(account)
+		require.Nil(t, err)
+		require.False(t, putCalled)
+	})
+
+	t.Run("old and new account already migrated, new code data, should update code in storer", func(t *testing.T) {
+		t.Parallel()
+
+		codeData := []byte("codeData")
+		newCodeData := []byte("newCodeData")
+
+		args := createMockAccountsDBArgs()
+		args.EnableEpochsHandler = &enableEpochsHandlerMock.EnableEpochsHandlerStub{
+			IsFlagEnabledCalled: func(flag core.EnableEpochFlag) bool {
+				return flag == common.MigrateCodeLeafFlag
+			},
+		}
+		newCodeDataHash := args.Hasher.Compute(string(newCodeData))
+
+		adr := make([]byte, 32)
+		oldAcc := stateMock.NewAccountWrapMock(adr)
+		oldAcc.SetCode(codeData)
+		oldAcc.SetVersion(uint8(core.WithoutCodeLeaf))
+
+		oldAccBytes, _ := args.Marshaller.Marshal(oldAcc)
+
+		putCalled := false
+
+		getCalls := 0
+
+		ts := &trieMock.TrieStub{
+			GetCalled: func(_ []byte) (common.TrieLeafHolder, error) {
+				if getCalls == 0 {
+					return common.NewTrieLeafHolder(oldAccBytes, 1, core.WithoutCodeLeaf), nil
+				}
+
+				getCalls++
+				return common.NewTrieLeafHolder(nil, 0, core.NotSpecified), nil
+			},
+			UpdateCalled: func(key, value []byte) error {
+				return nil
+			},
+			UpdateWithVersionCalled: func(key, value []byte, version core.TrieNodeVersion) error {
+				return nil
+			},
+			GetStorageManagerCalled: func() common.StorageManager {
+				return &storageManager.StorageManagerStub{
+					PutCalled: func(key, value []byte) error {
+						putCalled = true
+						require.Equal(t, key, newCodeDataHash)
+						require.Equal(t, value, newCodeData)
+						return nil
+					},
+				}
+			},
+		}
+
+		args.Trie = ts
+
+		adb, err := state.NewAccountsDB(args)
+		require.Nil(t, err)
+
+		adr = make([]byte, 32)
+		account := stateMock.NewAccountWrapMock(adr)
+		account.SetCode(newCodeData)
+		account.SetVersion(uint8(core.WithoutCodeLeaf))
+
+		err = adb.SaveAccount(account)
+		require.Nil(t, err)
+		require.True(t, putCalled)
+	})
+
+	t.Run("old account not migrated, should update old code references in trie and new code in storer", func(t *testing.T) {
```

Comment:
Add another case like this, but then revert the changes with `adb.RevertToSnapshot()`.

Reply:
added another test case